File: ex_007.py
# сортировка выбором (минимального элемента со смещением вправо)

arr = [6, 1, 2, 8, 10, 3, 7]

def selection_sort(arr):
    for i in range(len(arr)-1):
        # предполагаем, что в индексе i - минимальное число
        index_of_min_element = i
        
        # внутренний цикл для смешения 
        for j in range(i+1, len(arr)):
            if arr[j] < arr[index_of_min_element]:
                index_of_min_element = j
        
        # обмен значениями 
        if (index_of_min_element != i):
            arr[i], arr[index_of_min_element] = arr[index_of_min_element], arr[i]

    return arr


print(selection_sort(arr))

File: ex_007.js
// сортировка выбором

function selectionSort(arr) {
  const n = arr.length;

  // запускаем цикл по длине массива
  for (let i = 0; i < n - 1; i++) {
    // добавляем внутренний второй цикл, где на каждой итерации сдвигаемся вправо (j=i+1)

    let minIndex = i;

    for (let j = i + 1; j < n; j++) {
      if (arr[j] < arr[minIndex]) {
        minIndex = j;
      }
    }

    // ставим минимальный элемент в нужное место

    if (minIndex !== i) {
      let temp = arr[i];
      arr[i] = arr[minIndex];
      arr[minIndex] = temp;
    }
  }

  return arr;
}

let arr = [6, 1, 2, 8, 10, 3, 7];

console.log(selectionSort(arr));
